File: javastory.budgetsh.stage4/src/javastory/budgetsh/stage4/share/service/budget/CashBookService.java
package javastory.budgetsh.stage4.share.service.budget;

import java.util.ArrayList;

import javastory.budgetsh.stage4.share.domain.budget.budget.CashBook;

public interface CashBookService {

	public boolean exist(String bankAccount);

	public boolean regist(CashBook cashbook);

	public CashBook retrieve(String account);

	public void update(CashBook foundCashbook);

	public void remove(CashBook foundCashbook);

	public ArrayList<CashBook> retrieveAll();

}

File: javastory.budgetsh.stage5.1/src/step2/logic/CashbookService.js
const CashbookStore = require('../store/filestore/CashBookStore.js');

exports.exist = function(bankAccount) {
	if (CashbookStore.exist(bankAccount)) {
		return true;
	}
	return false;
}

exports.regist = function(cashbook) {
	let bankAccount = cashbook.bankAccount;
	if (CashbookStore.exist(bankAccount)) {
		return false;
	}
	CashbookStore.regist(cashbook);

	if (!CashbookStore.exist(bankAccount)) {
		return false;
	}
	return true;
}

exports.retrieve = function(bankAccount) {
	if (!CashbookStore.exist(bankAccount)) {
		return null;
	}
	return CashbookStore.retrieve(bankAccount);
}

exports.update = function(cashbook) {
	let bankAccount = cashbook.bankAccount;
	if (!CashbookStore.exist(bankAccount)) {
		return false;
	}
	CashbookStore.update(cashbook);
	return true;
}

exports.remove = function(cashbook) {
	let bankAccount = cashbook.bankAccount;
	if(!CashbookStore.exist(bankAccount)){
		return false;
	}
	CashbookStore.remove(cashbook);
	if(CashbookStore.exist(bankAccount)){
		return false;
	}
	return true;
}